base/uk.ac.stfc.isis.ibex.activemq/src/uk/ac/stfc/isis/ibex/activemq/SendReceiveSession.java

```diff
@@ -109,19 +110,36 @@ protected void throwingConnect() throws JMSException {
      * 
      * @param text
      *            The message to send.
+     * @return message details for sent message
      */
-    public void sendMessage(String text) {
-        TextMessage txtMessage;
+    public SendMessageDetails sendMessage(String text) {
+        if (session == null) {
+            return SendMessageDetails.createSendFail("No connection to script server.", "");
+        }
+
+        String messageId = getNextCorrelationID();
         try {
-            txtMessage = session.createTextMessage();
+            TextMessage txtMessage = session.createTextMessage();
             txtMessage.setText(text);
             txtMessage.setJMSReplyTo(tempQueue);
+            txtMessage.setJMSCorrelationID(messageId);
             producer.send(txtMessage);
+            return SendMessageDetails.createSendSuccess(messageId);
         } catch (JMSException e) {
             LOG.warn("Failed to send message " + text);
+            return SendMessageDetails.createSendFail(e.getLocalizedMessage(), messageId);
         }
     }
 
+    /**
+     * @return a new correlation ID
+     */
+    private String getNextCorrelationID() {
+        Random random = new Random(System.currentTimeMillis());
```

Comment:
Just needs to be different. We are using it to correlate messages not securing them.